app/src/main/java/com/group2/catan_android/GameActivity.java

```diff
@@ -148,58 +155,86 @@ public ImageView[] setupViews(ConstraintLayout constraintLayout, int totalViews,
             constraintLayout.addView(view, params);
             views[i] = view;
 
-            if(clickableElement != null){
-                setOnClickListener(view,correctID,clickableElement);
+            if (clickableElement != null) {
+                setOnClickListener(view, correctID, clickableElement);
             }
         }
 
         return views;
     }
 
-    private void setOnClickListener(View view, int correctID, ClickableElement clickableElement){
+    private void setOnClickListener(View view, int correctID, ClickableElement clickableElement) {
         view.setOnClickListener(v -> {
-                try {
-                    switch (clickableElement){
-                        case CONNECTION: clickOnConnection(correctID); break;
-                        case INTERSECTION: clickOnIntersection(correctID); break;
-                        case ROBBER: clickOnRobber(correctID); break;
-                        default: break;
-                    }
-                    currentButtonFragmentListener.onButtonEvent(lastButtonClicked);
-                    lastButtonClicked = null;
-                    gameEffectManager.playSound(R.raw.pop);
-                } catch (Exception e) {
-                    MessageBanner.makeBanner(this, MessageType.ERROR, e.getMessage()).show();
-                    gameEffectManager.playSound(R.raw.small_error);
-                    gameEffectManager.doubleVibrate();
+            try {
+                switch (clickableElement) {
+                    case CONNECTION:
+                        clickOnConnection(correctID);
+                        break;
+                    case INTERSECTION:
+                        clickOnIntersection(correctID);
+                        break;
+                    case ROBBER:
+                        clickOnRobber(correctID);
+                        break;
+                    default:
+                        break;
                 }
+                currentButtonFragmentListener.onButtonEvent(lastButtonClicked);
+                if (clickableElement != ClickableElement.ROBBER) lastButtonClicked = null;
```

Comment:
if statement is not necessary, if whatever move is succesfull in try block then lastButtonClicked should always be reseted to null, meaning that no button is active and a player has to select the button again to make a new move after a successfull move